File: Mis Prácticas/Exámenes/Examen 3/Napakalaki/src/napakalaki/Dice.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

import java.util.Random;

/**
 *
 * @author fran
 */
public class Dice {
    
    private static final Dice instance = new Dice();
    
    // Constructor privado
    private Dice(){
        
    }
    
    public static Dice getInstance(){
        return instance;
    }
    
    // Genera un número aleatorio entre 1 y 6 (ambos incluidos)
    public int nextNumber(){
        return new Random().nextInt(6)+1;
    }
    
}

File: Mis Prácticas/NapakalakiRuby/lib/dice.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require "singleton"

class Dice

  include Singleton
  
  private
  
  def initialize
    
  end
  
  private_class_method :new
  
  public
  
  def next_number
    rand(6)+1
  end
    
end
